benchmarks/experiment_runner.py

```diff
@@ -440,9 +446,26 @@ def parse_args(args=None):
   return parser.parse_args(args)
 
 
+FILTER_BY_TIER = {
+  1: r"^(BERT_pytorch|cm3leon_generate|DALLE2_pytorch|dlrm|hf_GPT2|hf_GPT2_large|GPT_3|hf_T5|hf_T5_base|hf_T5_generate|hf_T5_large|llama_v2_7b_16h|stable_diffusion_xl)$",
+  2: r"^(alexnet|attention_is_all_you_need_pytorch|Background_Matting|basic_gnn_gcn|basic_gnn_gin|basic_gnn_sage|dcgan|densenet121|detectron2_fasterrcnn_r_101_c4|detectron2_fasterrcnn_r_101_dc5|detectron2_fasterrcnn_r_101_fpn|detectron2_fasterrcnn_r_50_c4|detectron2_fasterrcnn_r_50_dc5|detectron2_fasterrcnn_r_50_fpn|detectron2_fcos_r_50_fpn|detectron2_maskrcnn|detectron2_maskrcnn_r_101_c4|detectron2_maskrcnn_r_101_fpn|detectron2_maskrcnn_r_50_c4|detectron2_maskrcnn_r_50_fpn|fastNLP_Bert|functorch_dp_cifar10|hf_Albert|hf_Bart|hf_Bert|hf_Bert_large|llama)$",
+  3: r"^(doctr_det_predictor|doctr_reco_predictor|drq|functorch_maml_omniglot)$",
+}
+
+
 def main():
   args = parse_args()
 
+  # Set filter based on tier
+  if args.filter_by_tier is not None:
+    args.filter = []
+    args.exclude = [r"^$"]
```

Comment:
I wouldn't overwrite args.filter nor args.exclude.
For example, one might want to run tier1, except BERT_pytorch (for example because it's known to not work). This could be accomplished with --filter-by-tier=1 --exclude=BERT_pytorch. Similarly, we might want to run tier1 plus a handful of benchmarks from other tiers, e.g. --filter-by-tier=1 --filter=hf_Bert. This should still be possible.

So, how about the following:

1. Set the filter and filter-by-tier options to a `[]` default (Note: [] also evaluates to false, just like None, so the code after this still works).
2. Replace this whole hunk with:
```
for tier in args.filter_by_tier:
  if tier not in FILTER_BY_TIER:
    continue
  args.filter.append(FILTER_BY_TIER[tier])
```

Reply:
Nice. Also added exclusion as discussed.

Reply:
Thanks. Still needs the default [] in the filter and exclude args, otherwise we might append to None, which cannot happen.
I'd also set the filter|exclude-by-tier args to a default [], so that we can directly iterate over them instead of checking for !None.

Reply:
Thanks!